editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java

```diff
@@ -0,0 +1,350 @@
+import com.esri.arcgisruntime.geometry.GeometryType;
+import com.esri.arcgisruntime.geometry.Part;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.geometry.Polyline;
+import com.esri.arcgisruntime.geometry.PolylineBuilder;
+import com.esri.arcgisruntime.geometry.ProximityResult;
+import com.esri.arcgisruntime.layers.AnnotationLayer;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.IdentifyLayerResult;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+public class EditFeaturesWithFeatureLinkedAnnotationSample extends Application {
+
+  private MapView mapView;
+  private FeatureLayer addressPointFeatureLayer;
+  private FeatureLayer parcelLinesFeatureLayer;
+  private AnnotationLayer addressPointsAnnotationLayer;
+  private AnnotationLayer parcelLinesAnnotationLayer;
+  private Feature selectedFeature = null;
+  private boolean selectedFeatureIsPolyline = false;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add JavaFX scene to stage
+      stage.setTitle("Edit Features With Feature-linked Annotation Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create the map with a light gray canvas basemap centered on Loudoun, Virginia
+      ArcGISMap map = new ArcGISMap(Basemap.Type.LIGHT_GRAY_CANVAS_VECTOR, 39.0204, -77.4159, 18);
+
+      // create a map view and set its map
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // add the map view to stack pane
+      stackPane.getChildren().addAll(mapView);
+      
+      // create and load the geodatabase
+      File geodatabaseFile = new File(System.getProperty("data.dir"),
+        "./samples-data/loudon/loudoun_anno.geodatabase");
+      Geodatabase geodatabase = new Geodatabase(geodatabaseFile.getAbsolutePath());
+
+      geodatabase.addDoneLoadingListener(() -> {
+        if (geodatabase.getLoadStatus() == LoadStatus.LOADED) {
+          // create feature layers from tables in the geodatabase
+          addressPointFeatureLayer = new FeatureLayer(geodatabase.getGeodatabaseFeatureTable("Loudoun_Address_Points_1"));
+          parcelLinesFeatureLayer = new FeatureLayer(geodatabase.getGeodatabaseFeatureTable("ParcelLines_1"));
+          // create annotation layers from tables in the geodatabase
+          addressPointsAnnotationLayer = new AnnotationLayer(geodatabase.getGeodatabaseAnnotationTable("Loudoun_Address_PointsAnno_1"));
+          parcelLinesAnnotationLayer = new AnnotationLayer(geodatabase.getGeodatabaseAnnotationTable("ParcelLinesAnno_1"));
```

Comment:
by the time we're using local variables for the FeatureLayer's here (see comment above), that's a lot of references to the word FeatureLayer in this block of code. For situations like this, we can make use of Java 10's keyword "var". This replaces the type information when declaring local variables and clears the readability up a little.

Also since these are quite long lines of code, we should stagger them a little:

 ```suggestion
          // create feature layers from tables in the geodatabase
          var addressPointFeatureLayer =
            new FeatureLayer(geodatabase.getGeodatabaseFeatureTable("Loudoun_Address_Points_1"));
          var parcelLinesFeatureLayer =
            new FeatureLayer(geodatabase.getGeodatabaseFeatureTable("ParcelLines_1"));
          // create annotation layers from tables in the geodatabase
          var addressPointsAnnotationLayer =
            new AnnotationLayer(geodatabase.getGeodatabaseAnnotationTable("Loudoun_Address_PointsAnno_1"));
          var parcelLinesAnnotationLayer =
            new AnnotationLayer(geodatabase.getGeodatabaseAnnotationTable("ParcelLinesAnno_1"));
```